test/integration/list_metadata_formats_test.rb

```diff
@@ -0,0 +1,35 @@
+require 'test_helper'
+
+class ListMetadataFormatsTest < ActionDispatch::IntegrationTest
+
+  include Oaisys::Engine.routes.url_helpers
+
+  setup do
+    @routes = Oaisys::Engine.routes
+  end
+
+  def test_list_metadata_formats_xml
+    get oaisys_path + '?verb=ListMetadataFormats', headers: { 'Accept' => 'application/xml' }
+    assert_response :success
+
+    schema = Nokogiri::XML::Schema(File.open(file_fixture('OAI-PMH.xsd')))
+    document = Nokogiri::XML(@response.body)
+    assert_empty schema.validate(document)
+
+    supported_formats = Oaisys::PMHController::SUPPORTED_FORMATS
```

Comment:
The danger with doing it this way is, our assertions are just reiterating our implementation, so it doesn't really "check our work" in the sense that if someone committed a typo in the controller, this test will pass because "typo" == "typo".

With the tests we want to unDRY ourselves a bit and repeat what we're expecting, so that we can catch any mismatch between the implementation and the expectation.